presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2217,14 +2240,12 @@ public ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession se
     @Override
     public ConnectorPartitioningHandle getPartitioningHandleForExchange(ConnectorSession session, int partitionCount, List<Type> partitionTypes)
     {
-        return new HivePartitioningHandle(
-                partitionCount,
-                partitionTypes.stream()
-                        .map(type -> toHiveType(
-                                typeTranslator,
-                                translateHiveUnsupportedTypeForTemporaryTable(type, typeManager)))
-                        .collect(toImmutableList()),
-                OptionalInt.empty());
+        if (!partitionTypes.isEmpty()) {
+            return createPartitioningHandleWithPrestoType(partitionCount, partitionTypes, OptionalInt.empty());
+        }
+        else {
+            return createPartitioningHandleWithHiveType(partitionCount, ImmutableList.of(), OptionalInt.empty());
```

Comment:
Why do we have to create Hive compatible partitioning if partitionTypes is empty?

Reply:
It doesn't matter which one to use,  let me use Presto Native to make code simple.